protocol/cors/enumerate-headers.feature

```diff
@@ -0,0 +1,22 @@
+Feature: Server should enumerate headers in Access-Control-Expose-Headers
+
+  Background: Set up test container (not empty)
+    * def testContainer = rootTestContainer.createContainer()
+    * def resource = testContainer.createResource('.txt', 'Hello', 'text/plain')
+
+  Scenario: Access-Control-Expose-Headers is present but not *
+    Given url testContainer.url
+    And headers clients.alice.getAuthHeaders('GET', testContainer.url)
+    And header Origin = 'https://tester'
+    And header Accept = 'text/turtle'
+    When method GET
+    Then status 200
+    And match header Access-Control-Expose-Headers != null
+    And match header Access-Control-Expose-Headers != '*'
+    And match response != ''
+
+  # To test this enumerates all headers we could split it on ', ' and compare to the set of all keys in responseHeaders
+  # The FETCH spec says
+  #   "A response will typically get its CORS-exposed header-name list set by extracting header values from the
+  #   `Access-Control-Expose-Headers` header. This list is used by a CORS filtered response to determine which headers to expose."
+  # This makes it sound as though the important thing is the intersection - it doesn't matter if headers are in one list but not the other.
```

Comment:
Not intersection, but rather that the set of returned headers is a subset of the exposed headers (minus CORS headers).

Reply:
In the example below, the exposed header list is: 
`Accept-Patch,ETag,Last-Modified,Link,Location,MS-Author-Via,Updates-Via,WAC-Allow`
but the returned headers also include the following which are neither CORS headers nor listed in exposed header list):
`X-Powered-By, Allow, Content-Type, Date, Connection, Keep-Alive, Transfer-Encoding`
Note that Location is allowed but not used so neither list is a subset of the other.

Therefore I think the CORS filtered response will include: `Accept-Patch,ETag,Last-Modified,Link,MS-Author-Via,Updates-Via,WAC-Allow` but none of the other headers. 

I subsequently found a reference to CORS safe headers which are also included in the filtered response and `Content-Type` is on that list so it is not required on the exposed header list (nor is `Last-Modified`). See https://fetch.spec.whatwg.org/#concept-filtered-response-cors

I think you can see my confusion over this. Each server I have tested has returned a different selection of headers in the exposed list so it is difficult to see how to test this beyond what I have done. Any ideas?

```
1 > GET https://server/alice/4B9dM7/eMG1w1/
1 > Authorization: DPoP ***EZHggw
1 > User-Agent: Solid-Conformance-Test-Suite
1 > DPoP: ***mpocjg
1 > Origin: https://tester
1 > Accept: text/turtle
1 > Host: server
1 > Connection: Keep-Alive
1 > Accept-Encoding: gzip,deflate

1 < 200
1 < Vary: Accept,Authorization,Origin
1 < X-Powered-By: Community Solid Server
1 < Updates-Via: wss://server/
1 < Access-Control-Allow-Origin: https://tester
1 < Access-Control-Allow-Credentials: true
1 < Access-Control-Expose-Headers: Accept-Patch,ETag,Last-Modified,Link,Location,MS-Author-Via,Updates-Via,WAC-Allow
1 < Accept-Patch: application/sparql-update
1 < Allow: OPTIONS, HEAD, GET, PATCH, POST, PUT, DELETE
1 < MS-Author-Via: SPARQL
1 < Content-Type: text/turtle
1 < Last-Modified: Mon, 31 Jan 2022 16:18:36 GMT
1 < ETag: "1643645916000"
1 < Link: <http://www.w3.org/ns/ldp#BasicContainer>; rel="type"
1 < Link: <http://www.w3.org/ns/ldp#Container>; rel="type"
1 < Link: <http://www.w3.org/ns/ldp#Resource>; rel="type"
1 < Link: <https://server/alice/4B9dM7/eMG1w1/.acl>; rel="acl"
1 < WAC-Allow: user="append control read write"
1 < Date: Mon, 31 Jan 2022 16:18:36 GMT
1 < Connection: keep-alive
1 < Keep-Alive: timeout=5
1 < Transfer-Encoding: chunked
```